go/tools/go-upgrade/go_upgrade.go

```diff
@@ -0,0 +1,498 @@
+/*
+Copyright 2023 The Vitess Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package main
+
+import (
+	"fmt"
+	"io"
+	"log"
+	"net/http"
+	"os"
+	"path"
+	"regexp"
+	"strconv"
+	"strings"
+	"time"
+
+	"encoding/json"
+
+	"github.com/hashicorp/go-version"
+	"github.com/spf13/cobra"
+)
+
+const (
+	goDevAPI = "https://go.dev/dl/?mode=json"
+)
+
+type latestGolangRelease struct {
+	Version string `json:"version"`
+	Stable  bool   `json:"stable"`
+}
+
+var (
+	workflowUpdate    = true
+	allowMajorUpgrade = false
+	isMainBranch      = false
+	goTo              = ""
+
+	rootCmd = &cobra.Command{
```

Comment:
Hey @mattlord, I think we are not seeing the same thing on our local environment. Running the different commands you listed above works as expected for me:

`go run ./go/tools/go-upgrade/go_upgrade.go --help`:
```
go-upgrade allows us to automate some tasks required to bump the version of Golang used throughout our codebase.

It mostly used by the update_golang_version.yml CI workflow that runs on a CRON.

This tool is meant to be run at the root of the repository.

Usage:
  go-upgrade [flags]
  go-upgrade [command]

Available Commands:
  completion  Generate the autocompletion script for the specified shell
  get         Command to get useful information about the codebase.
  help        Help about any command
  upgrade     upgrade will upgrade the Golang and Bootstrap versions of the codebase to the latest available version.

Flags:
  -h, --help   help for go-upgrade

Use "go-upgrade [command] --help" for more information about a command.
```

-----------

`go run ./go/tools/go-upgrade/go_upgrade.go upgrade --help`:
```
This command bumps the Golang and Bootstrap versions of the codebase.

The latest available version of Golang will be fetched and used instead of the old version.

By default, we do not allow major Golang version upgrade such as 1.20 to 1.21 but this can be overridden using the
--allow-major-upgrade CLI flag. Usually, we only allow such upgrade on the main branch of the repository.

In CI, particularly, we do not want to modify the workflow files before automatically creating a Pull Request to
avoid permission issues. The rewrite of workflow files can be disabled using the --workflow-update=false CLI flag.

Moreover, this command automatically bumps the bootstrap version of our codebase. If we are on the main branch, we
want to use the CLI flag --main to remember to increment the bootstrap version by 1 instead of 0.1.

Usage:
  go-upgrade upgrade [flags]
  go-upgrade upgrade [command]

Available Commands:
  workflows   workflows will upgrade the Golang version used in our CI workflows files.

Flags:
      --allow-major-upgrade   Defines if Golang major version upgrade are allowed.
  -h, --help                  help for upgrade
      --main                  Defines if the current branch is the main branch.
      --workflow-update       Whether or not the workflow files should be updated. Useful when using this script to auto-create PRs. (default true)

Use "go-upgrade upgrade [command] --help" for more information about a command.
```


----------
`go run ./go/tools/go-upgrade/go_upgrade.go get go-version`:
```
1.20.1
```

Reply:
Ahhh, my bad. I was confused myself as to why the behavior didn't seem to line up with the code I was looking at. It's because the PR branch is off of the `vitessio/vitess` repo but I had checked out a branch of the same name off the `planetscale/vitess` repo. We should delete that upstream branch to avoid any other confusion. Sorry about that! 

I can confirm that the output is good and what's expected. 🙂 I was wondering why it was not behaving like a Cobra app and why it didn't line up with the code I was reviewing... 🤣